lib/rules/at-rule-property-required-list/index.js

```diff
@@ -30,39 +30,60 @@ const rule = (primary) => {
 			return;
 		}
 
+		/** @type {Map<string, Set<string>|undefined>} */
+		const propLists = new Map();
+
+		for (const key in primary) {
+			if (!Object.hasOwnProperty.call(primary, key)) continue;
+
+			const propList = flattenArray(primary[key]);
+
+			if (!propList) continue;
+
+			propLists.set(key, new Set(propList));
+		}
```

Comment:
Should we create a utility for this?

`Record<string, string[] | string | undefined>` to `Map<string, Set<string>>`

Reply:
If you need it with later work, you can create a utility on this PR.

I don't think we have a utility to do the same work.

Reply:
There are 4 rules with similar option types (`Rule<Record<string, string | string[]>>`)
Looking at those I think it will come up again.

Do you have a name suggestion?

Reply:
Hum, how about `createMapWithSet()`? For example:

```js
/**
 * @template T
 * @param {Record<string, T | T[]>} records
 * @returns {Map<string, Set<T>>}
 */
function createMapWithSet(records) {
	// ...
}
```

Reply:
I like this!